source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2255,6 +2255,209 @@ void COpenGLDriver::SAuxContext::setActiveDescriptorSet(uint32_t index, const CO
+            glCullFace(nextState.rasterParams.cullFace);
+            UPDATE_STATE(rasterParams.cullFace);
+        }
+        if (STATE_NEQ(rasterParams.stencilTestEnable)) {
+            disable_enable_fptr[nextState.rasterParams.stencilTestEnable](GL_STENCIL_TEST);
+            UPDATE_STATE(rasterParams.stencilTestEnable);
+        }
+        if (nextState.rasterParams.stencilTestEnable && STATE_NEQ(rasterParams.stencilOp_front)) {
+            COpenGLExtensionHandler::extGlStencilOpSeparate(GL_FRONT, nextState.rasterParams.stencilOp_front.sfail, nextState.rasterParams.stencilOp_front.dpfail, nextState.rasterParams.stencilOp_front.dppass);
+            UPDATE_STATE(rasterParams.stencilOp_front);
+        }
+        if (nextState.rasterParams.stencilTestEnable && STATE_NEQ(rasterParams.stencilOp_back)) {
+            COpenGLExtensionHandler::extGlStencilOpSeparate(GL_BACK, nextState.rasterParams.stencilOp_back.sfail, nextState.rasterParams.stencilOp_back.dpfail, nextState.rasterParams.stencilOp_back.dppass);
+            UPDATE_STATE(rasterParams.stencilOp_back);
+        }
+        if (nextState.rasterParams.stencilTestEnable && STATE_NEQ(rasterParams.stencilFunc_front)) {
+            COpenGLExtensionHandler::extGlStencilFuncSeparate(GL_FRONT, nextState.rasterParams.stencilFunc_front.func, nextState.rasterParams.stencilFunc_front.ref, nextState.rasterParams.stencilFunc_front.mask);
+            UPDATE_STATE(rasterParams.stencilFunc_front);
+        }
+        if (nextState.rasterParams.stencilTestEnable && STATE_NEQ(rasterParams.stencilFunc_back)) {
+            COpenGLExtensionHandler::extGlStencilFuncSeparate(GL_FRONT, nextState.rasterParams.stencilFunc_back.func, nextState.rasterParams.stencilFunc_back.ref, nextState.rasterParams.stencilFunc_back.mask);
+            UPDATE_STATE(rasterParams.stencilFunc_back);
+        }
+        if (STATE_NEQ(rasterParams.depthTestEnable)) {
+            disable_enable_fptr[nextState.rasterParams.depthTestEnable](GL_DEPTH_TEST);
+            UPDATE_STATE(rasterParams.depthTestEnable);
+        }
+        if (nextState.rasterParams.depthTestEnable && STATE_NEQ(rasterParams.depthFunc)) {
+            glDepthFunc(nextState.rasterParams.depthFunc);
+            UPDATE_STATE(rasterParams.depthFunc);
+        }
+        if (STATE_NEQ(rasterParams.frontFace)) {
+            glFrontFace(nextState.rasterParams.frontFace);
+            UPDATE_STATE(rasterParams.frontFace);
+        }
+        if (STATE_NEQ(rasterParams.depthClampEnable)) {
+            disable_enable_fptr[nextState.rasterParams.depthClampEnable](GL_DEPTH_CLAMP);
+            UPDATE_STATE(rasterParams.depthClampEnable);
+        }
+        if (STATE_NEQ(rasterParams.rasterizerDiscardEnable)) {
+            disable_enable_fptr[nextState.rasterParams.rasterizerDiscardEnable](GL_RASTERIZER_DISCARD);
+            UPDATE_STATE(rasterParams.rasterizerDiscardEnable);
+        }
+        //if (STATE_NEQ(rasterParams.polygonOffsetEnable))
```

Comment:
Vulkan has no polygon offset? How are we going to do shadows?

Reply:
ofc it has, it's a todo because in OpenGL it's scattered on 3 glEnable-s

Reply:
lol

Reply:
did you get around to doing it?

Reply:
yea, im enabling/disabling all 3

Reply:
(in opengl theres 1 for each polygon mode: point, line, fill)